backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/nodes/EmptyDiagramNodeDescriptionProvider.java

```diff
@@ -177,4 +160,31 @@ private NodeTool addExistingElementsTool() {
                 .body(addExistingelements.build())
                 .build();
     }
+    
+    private NodeToolSection[] createToolSections(IViewDiagramElementFinder cache) {
+        var sections = new ArrayList<NodeToolSection>();
+        
+        GeneralViewDiagramDescriptionProvider.TOOL_SECTIONS.forEach((sectionName, elements) -> {
+            NodeToolSectionBuilder sectionBuilder = this.diagramBuilderHelper.newNodeToolSection()
+                    .name(sectionName)
+                    .nodeTools(this.createElementsOfToolSection(cache, elements));
+            sections.add(sectionBuilder.build());
+        });
+        
+        sections.add(addElementsToolSection(cache));
+        
+        return sections.toArray(NodeToolSection[]::new);
+    }
+    
+    private NodeTool[] createElementsOfToolSection(IViewDiagramElementFinder cache, List<EClass> elements) {
+        var nodeTools = new ArrayList<NodeTool>();
+
+        elements.forEach(definition -> {
```

Comment:
Could you rename this `definition` variable please? I don't think is named correctly.

Reply:
Changed